### DISCUSSION TITLE
Reset approval status for PRs after new changes are added

### DISCUSSION THREAD

**Discussion by adarean5** [scania-digital-design-system/tegel#579] [Ideas]
I pushed changes to a PR that was already approved and I noticed that the approval status persisted even after my changes were pushed. In my case, the changes were due to me rebasing my branch with the latest developed branch. Since this is an open-source project, someone could get the PR approved and then sneak in some last-minute minor changes that would not be reviewed. This would most likely be accidental, but it could also be intentional.

The one downside is that PRs would often need to be re-approved if the changes are caused by resolving minor conflicts with the target branch. On the other hand, even conflict resolution can introduce unintended changes that could be caught by a review.

I'm not that familiar with the PR settings in GitHub, but GitLab allows you to toggle whether or not approvals for PRs are reset after new changes are pushed, so perhaps GitHub has a similar setting.

**Comment by theJohnnyMe** [comment]
Good point, I will speak with team about it. It adds to security for sure, but it might slow us too. Lets see if there is an option that rebase od develop is not asking a need for re-review the code again.

**Reply by timrombergjakobsson** [reply]
I think this makes sense to have as a setting. As you said, this is a open source project, with quite a few users potentially affected if something would "sneak in". I think there is a setting called "Dismiss stale pull request approvals when new commits are pushed" that can be enabled. I think it is worth trying out, @theJohnnyMe!

**Comment by joao-goncalves-vwgds** [comment]
Hi @adarean5,

As suggested, we enabled the following option:

<img width="584" height="45" alt="image" src="https://github.com/user-attachments/assets/0a0ff41d-277a-4030-ade6-f6ec095091f3" />

Which means that any commit will remove previous approvals on the PR.

We took the liberty of also enabling this option:

<img width="658" height="52" alt="image" src="https://github.com/user-attachments/assets/c7564b5e-f7d1-4d1d-b14c-d4800344e7d1" />

Which further prevents the person that pushed the latest commit, from being one of the approvers.

The first option clearly adds an important security measure, to ensure that no PRs are merged with stale approvals.
For the second one, we are still unsure how it will impact our workflow, and will now do some tests on the next PRs, to see how it behaves.
If we find that it slows us down more than we can benefit from it, we will consider disabling it, since we already require at least 2 approvers, which slightly defeats its purpose.

Thanks for the suggestion, and please let us know if you find any hurdles with this new process 🙏
